### DISCUSSION TITLE
How to create the Dataloader

### DISCUSSION THREAD

**Discussion by chenyuqi990215** [pyg-team/pytorch_geometric#3358] [Q&A]
Hi,

I'm pretty new to the field of graph neural networks and pytorch geometric.

Each data point in my data consists of a graph with N point and E edges (e.g. features with dimension N*C and edge_index with dimension E*2), together with an input sequence (e.g. dimension 1*D). How could I create such a dataset and enable mini-batch sampling.  

Is it possible to solve the problem with rewriting the collate_fn while create DataLoader? 

(I would like to use GCNConv in torch_geometry, therefore I would like for each batch sample, features with dimension (B*N)*C, edge_indexs with dimension (B*E)*2 and sequences with dimension B*N)

Thank you in advance.

**Comment by rusty1s** [comment]
Saving your data as
```python
data = Data(x=x, edge_index=edge_index, seq=seq)
```
where
```python
x = ... [N, C]
edge_index = ... [2, E]
seq = [1, D]
```
should already give you a `Batch` object with shapes:
```python
x = ... [B * N, C]
edge_index = ... [2, B * E]
seq = [B, D]
```
which can directly be used as input to `GCNConv`. Hope this helps :)

**Reply by chenyuqi990215** [reply]
Pretty thank for your answer. I come up with another question. Since the graph is static (i.e. the graph is the same for all the data point), and also the graph is every large (i.e. 60000+ edges). Is there some better ways of data preparing?

**Reply by rusty1s** [reply]
A current workaround is to save `data` without `edge_index`, and instead save it as a separate file in `process` and load it in `__init__`:
```python
self.edge_index = torch.load(path_to_edge_index)
```
You can then override the `get()` method as follows:
```python
def get(self, idx):
    data = super().get(idx)
    data.edge_index = self.edge_index
    return data
```
